src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -901,12 +897,8 @@ private static String replaceUnpredictableValuesWithPlaceholders(String content)
                 // today's date
                 .replace(TimeHelper.formatDate(now).replace("/", "&#x2f;"), "${today}")
                 .replace(TimeHelper.formatDate(now), "${today}")
-                // now (used in comments last edited date) e.g. [Thu, 07 May 2015, 07:52:13 UTC]
-                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}:[0-9]{2} UTC", "\\${comment\\.date}")
-                // now (date, time)
-                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2} [AP]M", "\\${datetime\\.now}")
-                // dynamic feedback submission numbers
-                .replaceAll("(?s)<span class=\"submissionsNumber\".*?</span>", "<span class=\"submissionsNumber\" id=\"submissionsNumber\">\\${submissions\\.number}</span>")
+                // now (date, time) e.g. [Thu, 07 May 2015, 07:52 PM] or [Thu, 07 May 2015, 07:52:13 UTC]
+                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}( [AP]M|:[0-9]{2} UTC)", "\\${datetime\\.now}")
```

Comment:
3 is not entirely true. There can be some places where truncation can be applied and yet the test will still pass, but this is because we didn't provide hard-coded test cases (i.e those with long google IDs), not because of the standardized placeholders. For the places with standardized placeholders, I have guarded them with enough hard-coded test cases.

Anyway, this has been a good discussion, but has gone well off-tangent w.r.t. the actual issue: the date/time format for comments. Date/time format for comments easily score a "yes" for all the 3 queries above.